app/(client)/event-calendar/page.tsx

```diff
@@ -0,0 +1,208 @@
+import {
+  Command,
+  CommandEmpty,
+  CommandInput,
+  CommandItem,
+  CommandList,
+} from '@/components/ui/command';
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+} from '@/components/ui/select';
+
+const EventsList = [
+  {
+    value: 'first-year-day',
+    label: 'День перваша',
+  },
+  {
+    value: 'faculty-day',
+    label: 'День факультету',
+  },
+];
+
+const SearchForEvent = () => {
+  const [value, setValue] = useState('');
+  const [hasSetValue, setHasSetValue] = useState(false);
+
+  return (
+    <Command>
+      <CommandInput
+        placeholder="Шукати захід"
+        autoComplete="true"
+        value={value}
+        onValueChange={newValue => setValue(newValue)}
+        onClick={() => {
+          setHasSetValue(false);
+        }}
+      />
+      <CommandList className={hasSetValue ? 'hidden w-0 h-0' : 'block'}>
+        <CommandEmpty>Не вдалось знайти захід.</CommandEmpty>
+        {EventsList.map(i => (
+          <CommandItem
+            key={i.value}
+            value={i.label}
+            onSelect={currentValue => {
+              setValue(currentValue === value ? '' : currentValue);
+              setHasSetValue(true);
+            }}
+          >
+            {i.label}
+          </CommandItem>
+        ))}
+      </CommandList>
+    </Command>
+  );
+};
+
+export default function EventCalendar() {
+  const [isDesc, setIsDesc] = useState(false);
+
+  function changeIcon() {
+    setIsDesc(!isDesc);
+  }
+
+  return (
+    <main className="flex flex-col items-center justify-center">
+      <div className="w-11/12">
+        <h1 className="text-5xl font-semibold max-[1280px]:text-lg">
+          Розклад заходів
+        </h1>
+        <div className="my-12 flex flex-row gap-5 max-[1280px]:flex-col">
+          <div className="w-1/4 max-[1280px]:w-full">
+            <SearchForEvent />
+          </div>
+          <div className="flex flex-row gap-3 h-14 max-h-14">
+            <Select>
+              <SelectTrigger className="border-zinc-200 w-1/8 max-w-1/4 rounded-xl">
+                Сортувати за
+              </SelectTrigger>
+              <SelectContent>
+                <SelectItem value={'time'}>Час</SelectItem>
+                <SelectItem value={'name'}>Назва</SelectItem>
+              </SelectContent>
+            </Select>
+            <Button
+              onClick={changeIcon}
+              variant={'outline'}
+              size={'sm'}
+              className="border-zinc-200 rounded-2xl"
+            >
+              {isDesc ? (
+                <SortAsc className="scale-150" />
+              ) : (
+                <SortDesc className="scale-150" />
+              )}
+            </Button>
+          </div>
+        </div>
+        <div className="w-full flex items-center justify-center max-[1280px]:px-3">
+          <div className="grid grid-cols-5 gap-5 gap-y-10 w-full max-[1280px]:flex max-[1280px]:flex-col max-[1280px]:gap-y-3">
```

Comment:
too soon the grid turns into a vertical list, at intermediate widths you can make a grid of 2-4 elements in a row